lib/ts/xmlhttprequest.ts

```diff
@@ -46,6 +46,11 @@ const XHR_EVENTS = [
 ] as const;
 
 export function addInterceptorsToXMLHttpRequest() {
+    // This is to prevent errors in server side environments where XMLHttpRequest is not defined
+    if (typeof XMLHttpRequest === "undefined") {
```

Comment:
Please add a similar check to all interceptors that return early (after reporting in debug logs that it'll skip adding the interceptor) in an SSR environment. Node 18 added a global `fetch`, which we are overriding if with this branch, with unintended side-effects.